soroban-sdk/src/error.rs

```diff
@@ -0,0 +1,33 @@
+use core::convert::Infallible;
+
+use crate::xdr;
+
+/// InvokeError captures errors returned from the invocation of another
+/// contract.
+#[derive(Debug, Copy, Clone, Eq, PartialEq)]
+pub enum InvokeError {
+    /// Abort occurs if the invoked contract function exited due to some
+    /// irrecoverable error, such as exceeding budget, or out-of-bounds access.
```

Comment:
That's not quite correct. The real irrecoverable errors (such as exceeding budget) aren't observable by the guest, as they cause the whole invocation to fail. I don't think we need to document that though (i.e. I wouldn't mention exceeding budget at all). I think a fair description would be that this covers any type of trap in the executed contract, either induced by an explicit `panic!`, or a host function failure, or a runtime error (like out-of-bounds access mentioned here).

Reply:
Thanks you're right, this is entirely wrong. I'll change it.